test/e2e/aws_test.go

```diff
@@ -19,112 +19,438 @@ limitations under the License.
+		createNamespace(namespace)
+
+		clusterName = "test-" + util.RandomString(6)
+		awsClusterName = "test-infra-" + util.RandomString(6)
+		cpMachinePrefix = "test-" + util.RandomString(6)
+		cpAWSMachinePrefix = "test-infra-" + util.RandomString(6)
+		cpBootstrapConfigPrefix = "test-boot-" + util.RandomString(6)
 	})
 
-	Describe("control plane node", func() {
-		It("should be running", func() {
-			sess := getSession()
-			accountID := getAccountID(sess)
+	Describe("workload cluster lifecycle", func() {
+		It("It should be creatable and deletable", func() {
+			By("Creating an AWSCluster")
+			makeAWSCluster(namespace, awsClusterName)
+
+			By("Creating a Cluster")
+			makeCluster(namespace, clusterName, awsClusterName)
+
+			By("Ensuring Cluster Infrastructure Reports as Ready")
+			waitForClusterInfrastructureReady(namespace, clusterName)
+
+			By("Creating the initial Control Plane Machine")
+			awsMachineName := cpAWSMachinePrefix + "-0"
+			bootstrapConfigName := cpBootstrapConfigPrefix + "-0"
+			machineName := cpMachinePrefix + "-0"
+			createInitialControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Deploying CNI to created Cluster")
+			deployCNI(testTmpDir, namespace, clusterName, *cniManifests)
+			waitForMachineNodeReady(namespace, machineName)
+
+			By("Creating the second Control Plane Machine")
+			awsMachineName = cpAWSMachinePrefix + "-1"
+			bootstrapConfigName = cpBootstrapConfigPrefix + "-1"
+			machineName = cpMachinePrefix + "-1"
+			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
 
-			createKeyPair(sess)
-			createIAMRoles(sess, accountID)
+			By("Creating the third Control Plane Machine")
+			awsMachineName = cpAWSMachinePrefix + "-2"
+			bootstrapConfigName = cpBootstrapConfigPrefix + "-2"
+			machineName = cpMachinePrefix + "-2"
+			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
+
+			// TODO: Deploy a MachineDeployment
+			// TODO: Scale MachineDeployment up
+			// TODO: Scale MachineDeployment down
+
+			By("Deleting the Cluster")
+			deleteCluster(namespace, clusterName)
 
-			namespace := "test-" + util.RandomString(6)
-			createNamespace(kindCluster.KubeClient(), namespace)
 		})
 	})
 })
 
-func createNamespace(client kubernetes.Interface, namespace string) {
-	_, err := client.CoreV1().Namespaces().Create(&corev1.Namespace{
+func deployCNI(tmpDir, namespace, clusterName, manifestPath string) {
+	cluster := &clusterv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: namespace,
+			Namespace: namespace,
+			Name:      clusterName,
 		},
-	})
-	Expect(err).To(BeNil())
+	}
+	kubeConfigData, err := kubeconfig.FromSecret(kindClient, cluster)
+	Expect(err).NotTo(HaveOccurred())
+	kubeConfigPath := path.Join(tmpDir, clusterName+".kubeconfig")
+	err = ioutil.WriteFile(kubeConfigPath, kubeConfigData, 0640)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
The kubeconfig part feels like almost out of place for this function, what if we extrapolate this out?

Reply:
Probably, I'd rather defer that until we need to rather than hold up this PR longer, though.